terraform/main.tf

```diff
@@ -125,6 +125,16 @@ module "arpa_treasury_report_security_group" {
   allow_all_egress = true
 }
 
+module "grant_digest_scheduled_task_security_group" {
+  source  = "cloudposse/security-group/aws"
+  version = "2.2.0"
+
+  namespace        = var.namespace
+  vpc_id           = data.aws_ssm_parameter.vpc_id.value
+  attributes       = ["grant_digest_scheduled_task"]
+  allow_all_egress = true
+}
+
```

Comment:
It occurs to me that, since the `grant_digest_scheduled_task` module defined in `terraform/modules/gost_api/grant_digest.tf` is configured with all the same security groups that are passed in for any other task (specifically the main API workers) controlled by `terraform/modules/gost_api`, this security group is redundant.

To be super verbose: the `terraform/modules/gost_api` module source uses all provided values in `var.security_group_ids` when placing tasks in various security groups. More to the point, it passes this same list directly to its `module "grants_digest_scheduled_task"` resource. Since this list already contains a security group  common to our Postgres cluster*, we don't need to provide another. The only reason this new security group would be necessary is if `terraform/modules/gost_api` declared a new input variable like `var.grants_digest_scheduled_task_security_groups` and _only_ used those values when configuring networking for that task.

I should have caught this during our review yesterday, but I was (mistakenly) thinking that the `module "grant_digest_scheduled_task"` resource would be defined in the main project (`terraform/` directory) rather than as a sub-resource of `module "gost_api"`. 

*Feel free to ignore everything below, but looking at the current security group configuration, it appears we somehow mismatched the `api_to_postgres_security_group` and `consume_grants_to_postgres_security_group` usages. The configuration _should_ be:
- `module.api_to_postgres_security_group.id` passed to:
  - `module "gost_api"` via `security_group_ids` input variable array
  - `module "postgres"` via `ingress_security_groups` input variable array
- `module.consume_grants_to_postgres_security_group.id` passed to the `security_group_ids` input variable arrays for:
  - `module "consume_grants"` via `security_group_ids` input variable array
  - `module "postgres"` via `ingress_security_groups` input variable array

This mismatch isn't the most major problem in the world – both security groups are configured identically, so this is ultimately just a naming convention, and they could even be consolidated down to a single security group. Just mentioning here because I noticed while reviewing.

**TL;DR** You can safely remove the `module "grant_digest_scheduled_task_security_group"` resource and its usages.

Reply:
Yeah, I think I was borrowing this idiom from things like `arpa_treasury_report_security_group`, but your assessment makes sense, and it does seem like consolidating security groups to a clear set of logical groupings, rather than "one per process", makes more sense to me. 

>I should have caught this during our review yesterday, but I was (mistakenly) thinking that the module "grant_digest_scheduled_task" resource would be defined in the main project (terraform/ directory) rather than as a sub-resource of module "gost_api".

Fwiw, it was located in `terraform/main.tf` originally, but I moved it under the `gost_api` module to reuse the `aws_ecs_task_definition` instead of having to redefine it. 

Anyway, I'll remove the new security group!